### ISSUE TITLE
Compilation ERROR, function(s)/method(s) not created! cc1plus.exe: error

### ISSUE THREAD

**Issue by CharlesXin** [stan-dev/rstan#307]
#### Summary:

"Compilation ERROR, function(s)/method(s) not created! cc1plus.exe: error: 
#### Description:

I have been getting the error like below. Totally have no idea,  the code work well yesterday but I haven't changed anything. May be due to the new version of rstan? 
#### Current Output:

Error in { : 
  task 1 failed - "Compilation ERROR, function(s)/method(s) not created! cc1plus.exe: error: unrecognized command line option '-ffat-lto-objects'
cc1plus.exe: warning: unrecognized command line option "-Wno-unused-local-typedefs" [enabled by default]
make: **\* [file3c307f925e20.o] Error 1
#### Expected Output:

Shouldn't have any error
#### RStan Version:

I tried both 2.10.1 and 2.9.0
#### R Version:

3.31 and even the older version. 
#### Operating System:

Windows 7

**Comment by bgoodri** [comment]
```
"Compilation ERROR, function(s)/method(s) not created! cc1plus.exe: error: unrecognized command line option '-ffat-lto-objects'
```

Your version of g++ does not support the `-ffat-lto-objects`. But if you are using R 3.3.x, then it requires RTools33, which includes version 4.9 of g++, which does support that flag. So my guess is that the real problem is that R is not finding version 4.9 of g++. There are instructions on how to configure RTools33 on https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started .

**Comment by aadler** [comment]
Just a warning about lto. You can compile base R (including recommended packages) with LTO if you know how to play with Makefile, Mkrules.dist, etc. but various other packages don't like LTO. For example, in my experience, dplyr will compile but hang if you successfully pass LTO (although that requires entering src/gnuwin32/fixed/etc) so as a rule, on Windows, I tend not to use it (outside of the original compilation of base R).

**Comment by aadler** [comment]
Honestly, for base R, it doesn't even add that much. See http://www.avrahamadler.com/2016/03/31/updated-r-blas-timings/

**Comment by bgoodri** [comment]
I have never looked at speed but LTO makes rstan.so much smaller, particularly with C++11 and / or -g.

**Comment by aadler** [comment]
Probably because you are not passing -ffat-lto-objects. Whenever I've tried building with LTO I've pass ed it, as not every package can be built on Windows with LTO, and without that flag, there may be issues trying to link a non-LTO object to an LTO object, at least according to my understanding of the gcc documentation.

**Comment by CharlesXin** [comment]
Hi guys! 

Thanks very much for your reply. I reinstalled the Rtool 3.3, but got a new error like below: 
task 1 failed - "Compilation ERROR, function(s)/method(s) not created! g++.exe: error: Files/R/R-3.3.1/library/StanHeaders/libs/x64: No such file or directory"

Any insights? 

Thanks very much!!!!!!!!!!!!!
Charles Xin

**Comment by bob-carpenter** [comment]
Sending your sessionInfo() can be really helpful.  And the full
message you get based on what you're trying to do.  
- Bob

**Comment by bgoodri** [comment]
It is the same space in the directory thing. The best thing to do is to install RStan from GitHub.

https://groups.google.com/d/msg/stan-users/xtP4UXIeUUw/LsZHJU6tCgAJ

**Comment by bgoodri** [comment]
This is fixed in 2.11.1
